website/blog/2024-06-26-llm-as-judge/index.md

```diff
@@ -0,0 +1,522 @@
+```python
+%env OPENAI_API_KEY=<your-api-key>
+```
+
+Import the necessary libraries
+```python
+import mlflow
+import os
+# Run a quick validation that we have an entry for the OPEN_API_KEY within environment variables
+assert "OPENAI_API_KEY" in os.environ, "OPENAI_API_KEY environment variable must be set"
+
+import openai
+import pandas as pd
+```
+
+When using mlflow.evaluate(), your LLM can be either:
+1.	A `mlflow.pyfunc.PyFuncModel()` . Generally speaking an MLflow model. 
+2.	A `python function` that has strings as inputs and outputs a single string. 
+3.	An `MLflow Deployments` endpoint URI.
+4.	Set `model=None` when you only want to provide the data which has already been scored by the model.
+
+For this case let's use an mlflow model.
+
+Let's start by logging our translation model on mlflow.
+For this tutorial let's use a gpt3.5 with a system prompt
+
+```python
+
+system_prompt = "Translate the following sentences into Spanish"
+basic_translation_model = mlflow.openai.log_model(
+    model="gpt-3.5-turbo",
+    task=openai.chat.completions,
+    artifact_path="model",
+    messages=[
+        {"role": "system", "content": system_prompt},
+        {"role": "user", "content": "{user_input}"},
+    ],
+)
+```
+
+Testing the model to make sure it works
+
+```python
+model = mlflow.pyfunc.load_model(basic_translation_model.model_uri)
+
+model.predict("Hello, how are you?")
+
+# Output = ['¡Hola, ¿cómo estás?']
+```
+
+To use mlflow evaluate we need to prepare some data to provide as input to our LLM.
+In this example we will define some common english expressions which we want translated.
+
+```python
+# 1. Prepare evaluation data
+eval_data = pd.DataFrame(
+    {
+        "llm_inputs": [
+            "I'm over the moon about the news!",
+            "Spill the beans.",
+            "Bite the bullet.",
+            "Better late than never.",
+            
+        ]
+    }
+)
+```
+
+
+This example demonstrates how to create a custom metric that uses an LLM to judge the cultural sensitivity of translations. The LLM considers context, idiomatic expressions, and cultural nuances that might be missed by traditional metrics.
+
+To achieve this let's define some custom metric. 
+We want to adress how faithfull the translation is, for that we have to consider cultural factors and not only a literal translation.
+
+Let's set a metric that takes into account that cultural sensitivity.
+
+By default MLflow evaluate uses openai:/gpt-4 as a judge. However you can choose also choose a local model to do this evaluation (for example using ollama wraped in a pyfync)
+
+```python
+#To use an endpoint hosted by a local MLflow Deployments Server, you can use the following code.
+
+from mlflow.deployments import set_deployments_target
+
+set_deployments_target("http://localhost:5000")
+my_answer_similarity = mlflow.metrics.genai.answer_similarity(
+    model="endpoints:/my-endpoint"
+)
+
+```
```

Comment:
I think it's OK to leave this out and keep focused on the gpt4 example, can just link to the docs